Source/RetryPolicy.swift

```diff
@@ -309,31 +309,49 @@ open class RetryPolicy: RequestInterceptor {
                     for session: Session,
                     dueTo error: Error,
                     completion: @escaping (RetryResult) -> Void) {
-        if
-            request.retryCount < retryLimit,
-            let httpMethod = request.request?.method,
-            retryableHTTPMethods.contains(httpMethod),
-            shouldRetry(response: request.response, error: error) {
-            let timeDelay = pow(Double(exponentialBackoffBase), Double(request.retryCount)) * exponentialBackoffScale
-            completion(.retryWithDelay(timeDelay))
+        if request.retryCount < retryLimit, shouldRetry(request: request, dueTo: error) {
+            if let timeDelay = delayForRetrying(request: request, dueTo: error) {
+                completion(.retryWithDelay(timeDelay))
+            } else {
+                completion(.retry)
+            }
         } else {
             completion(.doNotRetry)
         }
     }
 
-    private func shouldRetry(response: HTTPURLResponse?, error: Error) -> Bool {
-        if let statusCode = response?.statusCode, retryableHTTPStatusCodes.contains(statusCode) {
+    // MARK: - ShouldRetry
+
+    /// - Parameters:
+    ///     - request: `Request` that failed due to the provided `Error`.
+    ///     - error:    `Error` encountered while executing the `Request`.
+    /// - Returns: `true` if `request` HTTP method is contained in the retryable HTTP methods and `request` response or `error` contain any of the retryable statuses and codes.
+    open func shouldRetry(request: Request, dueTo error: Error) -> Bool {
+        guard let httpMethod = request.request?.method, retryableHTTPMethods.contains(httpMethod) else {
+            return false
+        }
+        if let statusCode = request.response?.statusCode, retryableHTTPStatusCodes.contains(statusCode) {
             return true
         } else {
             let errorCode = (error as? URLError)?.code
             let afErrorCode = (error.asAFError?.underlyingError as? URLError)?.code
-            if let code = errorCode ?? afErrorCode, retryableURLErrorCodes.contains(code) {
-                return true
-            } else {
+            guard let code = errorCode ?? afErrorCode else {
                 return false
             }
+            return retryableURLErrorCodes.contains(code)
         }
     }
+
+    // MARK: - DelayForRetrying
+
+    /// - Parameters:
+    ///     - request: `Request` that failed due to the provided `Error`.
+    ///     - error:    `Error` encountered while executing the `Request`.
+    /// - Returns: A  delay in seconds or nil to retry without a delay.
+    open func delayForRetrying(request: Request, dueTo error: Error) -> TimeInterval? {
```

Comment:
It doesn't make sense to allow overriding of the delay calculation, as `RetryPolicy` was designed around geometric backoff and doesn't really make sense without it.